### ISSUE TITLE
Deleted records appear in search results

### ISSUE THREAD

**Issue by chris-asl** [inspirehep/inspire-next#2933]
Deleted records appear on the normal search.

## Expected Behavior
They shouldn't, unless explicitly specified. 

## Steps to Reproduce (for bugs)
1. migrate this record https://labs.inspirehep.net/api/literature/1297774
`inspirehep migrator one --recid=1297774`
2. Do an empty search and you can find it.

### Notes:
1. If I write `deleted:true` as a query I can see only that record. But if I write `deleted:false` doesn't return anything.  
Same happens with ElasticSearch query 
```
GET records-hep/_search
{
  "query": {
    "match": {
        "deleted": true
      }
  }
} 
```
(either `true` or `false`, same behaviour as described above).

2. This [code](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/config.py#L148) seems related and it's being used [here](https://github.com/inveniosoftware/invenio-collections/blob/b120e1461884a79afb994be4c95703a02edb578d/invenio_collections/receivers.py#L52).

**Comment by david-caro** [comment]
This is closely related to the fact that the deleted records are considered as 'existing' by the workflows here: https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/tasks/matching.py#L98

supposedly because the matcher does not filter the deleted ones out.

**Comment by jacquerie** [comment]
> supposedly because the matcher does not filter the deleted ones out.

All you need to do this is to implement a `validator` that filters them. It's already supported by the config API!

**Comment by david-caro** [comment]
But the validator is not using the elasticsearch filters right? Is just a python function that is run after right?

**Comment by jacquerie** [comment]
> Is just a python function that is run after right?

Yes, that's right. We could easily add a boolean flag to the configuration that expands to a `filter` clause, though.
